runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -573,12 +574,331 @@ static TR::Instruction *generatePrefetchAfterHeaderAccess(TR::Node
+      //
+      loadHighInstr = generateRegMemInstruction(L4RegMem, node, highReg,
+                                                generateX86MemoryReference(*convertedLongMR, 4, cg), cg);
+
+      loadLowInstr = generateRegMemInstruction(L4RegMem, node, lowReg,
+                                               generateX86MemoryReference(*convertedLongMR, 0, cg), cg);
+
+      // Jump to the snippet if the converted value is an indefinite integer; otherwise continue.
+      //
+      generateRegImmInstruction(CMP4RegImm4, node, highReg, INT_MIN, cg);
+      generateLabelInstruction(JNE4, node, reStartLabel, cg);
+      generateRegRegInstruction(TEST4RegReg, node, lowReg, lowReg, cg);
+      generateLabelInstruction(JE4, node, snippetLabel, cg);
+
+      // Create the conversion snippet.
+      //
+      cg->addSnippet( new (cg->trHeapMemory()) TR::X86FPConvertToLongSnippet(reStartLabel,
+                                                                             snippetLabel,
+                                                                             helperSymRef,
+                                                                             node,
+                                                                             loadHighInstr,
+                                                                             loadLowInstr,
+                                                                             cg) );
+
+      // Make sure the high and low long registers are assigned to something.
+      //
+      if (accReg)
+         {
+         deps = generateRegisterDependencyConditions((uint8_t)0, 3, cg);
+         deps->addPostCondition(accReg, TR::RealRegister::eax, cg);
+         deps->addPostCondition(lowReg, TR::RealRegister::NoReg, cg);
+         deps->addPostCondition(highReg, TR::RealRegister::NoReg, cg);
+         }
+      else
+         {
+         deps = generateRegisterDependencyConditions((uint8_t)0, 2, cg);
+         deps->addPostCondition(lowReg, TR::RealRegister::NoReg, cg);
+         deps->addPostCondition(highReg, TR::RealRegister::NoReg, cg);
+         }
+      generateLabelInstruction(LABEL, node, reStartLabel, deps, cg);
+
+      // We want the floating point register to be live through the snippet, so if it is
+      // not referenced again we must pop it off the stack here.
+      //
+      if ((cg->decReferenceCount(child) == 0) || (child->getRegister()->getKind() == TR_FPR))
+         {
+         generateInstruction(FSTPST0, node, cg);
+         }
+
+      TR::Register *targetRegister = cg->allocateRegisterPair(lowReg, highReg);
+      node->setRegister(targetRegister);
+      return targetRegister;
+      }
+   }
+
+TR::Register *J9::X86::TreeEvaluator::f2iEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   bool doubleSource;
+   bool longTarget;
+   TR::InstOpCode::Mnemonic cvttOpCode;
+   // On AMD64, all four [fd]2[il] conversions are handled here
+   // On IA32, both [fd]2i conversions are handled here
```

Comment:
Can you move these two comments outside the function to make it clear what operations this function supports?  Otherwise, the reader has to begin to grok this function to discover that.